google/resource_compute_address.go

```diff
@@ -74,11 +98,36 @@ func resourceComputeAddressCreate(d *schema.ResourceData, meta interface{}) erro
 	}
 
 	// Build the address parameter
-	addr := &compute.Address{Name: d.Get("name").(string)}
-	op, err := config.clientCompute.Addresses.Insert(
-		project, region, addr).Do()
-	if err != nil {
-		return fmt.Errorf("Error creating address: %s", err)
+	addr := &computeBeta.Address{
+		Name:        d.Get("name").(string),
+		AddressType: d.Get("address_type").(string),
+		Subnetwork:  d.Get("subnetwork").(string),
+	}
+
+	var op interface{}
+	switch computeApiVersion {
+	case v1:
+		v1Address := &compute.Address{}
+		err = Convert(addr, v1Address)
+		if err != nil {
+			return err
+		}
+		op, err = config.clientCompute.Addresses.Insert(
+			project, region, v1Address).Do()
+		if err != nil {
```

Comment:
If you wanted to, you could move this block after the switch case since it is the same for beta and v1